Document non-default property values (#1605)
redpanda-data/documentation: docs/develop/kafka-clients.mdx

* Record deletion.

If you have any issues while working with a Kafka tool, you can [file an issue](https://github.com/redpanda-data/redpanda/issues/new).

> ## Non-default property values

Author: SMEs - please review for correctness and any properties missing from this table.

> Redpanda sets default values for the following properties that are different than Kafka's default values for corresponding properties:

Reviewer: ```suggestion
Redpanda sets default values for the following properties, which are different from Kafka's default values for corresponding properties:
```

| Property | Redpanda default value | Kafka default value | Notes |
| :-- | :-- | :-- | :-- |
|[`auto_create_topics_enabled`](../../reference/cluster-properties#auto_create_topics_enabled)| `false` | `true` | Prevents errant applications from creating too many new topics. | 

> |[`log_segment_ms`](../../reference/cluster-properties#log_segment_ms)| 14 days | 7 days | Sets the range of time within a segment that Redpanda can handle out-of-order timestamps. | 

Reviewer 2: The description for log_segment_ms doesn't match the property:
```
 , log_segment_ms(
      *this,
      "log_segment_ms",
      "Default log segment lifetime in ms for topics which do not set "
      "segment.ms",
      {.needs_restart = needs_restart::no,
       .example = "3600000",
       .visibility = visibility::user},
      std::chrono::weeks{2},
      {.min = 60s})

```

Author: @jcsp @VladLazar - the description in the table for `log_segment_ms` is meant to convey the change in its default value (from 7 days to 14 days) made in https://github.com/redpanda-data/redpanda/pull/8509. How do you recommend we revise or remove this description from the table?
